unraid_mcp/tools/system.py

```
@@ -197,6 +207,107 @@ def analyze_disk_health(disks: list[dict[str, Any]], disk_type: str) -> dict[str
+            swap_percent = mem.get('percentSwapTotal')
+            if swap_percent is not None:
+                summary['swap_usage'] = f"{swap_percent:.1f}%"
+            summary['swap_total'] = format_bytes(mem.get('swapTotal'))
+            summary['swap_used'] = format_bytes(mem.get('swapUsed'))
+            summary['swap_free'] = format_bytes(mem.get('swapFree'))
+
+        return {"summary": summary, "details": raw_metrics}
+
+    except Exception as e:
+        logger.error(f"Error in get_metrics: {e}", exc_info=True)
+        raise ToolError(f"Failed to retrieve system metrics: {str(e)}") from e
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)